client/team.go

```diff
@@ -81,3 +81,16 @@ func (client *ApiClient) Teams() ([]Team, error) {
 	}
 	return result, err
 }
+
+func (client *ApiClient) TeamsByName(name string) ([]Team, error) {
```

Comment:
To avoid code duplication, I recommend creating a helper function that receives a map.
For Teams() pass an empty map.
For TeamsByName() pass the name: name map.

Reply:
done in https://github.com/env0/terraform-provider-env0/pull/521/commits/01343c92fb08adfc82d356fd38ad98eec05011cc